lib/cocoapods/config.rb

```diff
@@ -100,11 +100,19 @@ def cache_root
     def initialize(use_user_settings = true)
       configure_with(DEFAULTS)
 
+      unless ENV['CP_HOME_DIR'].nil?
+        @cache_root = home_dir + 'cache'
+      end
+
       if use_user_settings && user_settings_file.exist?
         require 'yaml'
         user_settings = YAML.load_file(user_settings_file)
         configure_with(user_settings)
       end
+
+      unless ENV['CP_CACHE_DIR'].nil?
+        @cache_root = Pathname.new(ENV['CP_CACHE_DIR']).expand_path
+      end
```

Comment:
Any reason for splitting this into two sections?

This will also override any values set by the yaml config file if it exists. If that's the desired behavior, we should have a test for that and document it as well.

Reply:
The reason is to allow overriding the config.

The first case could actually go into the `DEFAULTS` but then it becomes untestable because its static.

Then we load your config file, but `CP_CACHE_DIR` can still override the conf, that seemed like the most idiomatic way to combine config file and env vars.

Ill add a test to show that this is the case.